### DISCUSSION TITLE
Bettestack Logs for Symfony

### DISCUSSION THREAD

**Discussion by landsman** [symfony/symfony#57747] [Q&A] [answered]
Hello guys,

For our clients we want to use [Betterstack](https://betterstack.com/logs) on theirs projects and this is the first try to do so, show them.

I want to ask you if you can help me with Symfony implementation.
It is working with monolog fine, but there is one thing that I don't like about.

**Logs in Live tail are not formatted - there are not replaced variables.**

See my screenshot:

![Screenshot 2024-07-17 at 11 31 55](https://github.com/user-attachments/assets/07afd225-4522-4899-b14c-7a2832a73b6a)

It's looks like this:

![image](https://github.com/user-attachments/assets/e9524792-dd4b-461a-bb74-d1ff339937bc)

Do you hava an idea how to solve this? 

**Here is demo project** of the implementation here: https://github.com/landsman/betterstack-symfony/tree/v5.4-php-7 to reproduce it. 

You can easily run it via docker.

I followed doc: https://betterstack.com/docs/logs/php/, but this is probably more Symfony framework specific thing.

It would be very helpful if you could direct me how to improve it.

**Comment by ostrolucky** [comment, marked as answer]
This is standard monolog behaviour: placeholders are not replaced by default for couple reasons. First, these values are not lost, they are still shipped as part of log context (you can still see them in your first screenshot). Second, if you do replace them, you make it hard for you to search for concrete type of message, because now messages are dynamic so you are forced to use regex/wildcards. Compare this with default approach, where you just search literally for `Matched route "{route}"` message and as additional filter `context.route: index`. There are more reasons but I don't want to write essays on this.

If you don't like this behaviour, this is of course configurable, you just need to enable PSR-3 processor. It should work if you add following line at  https://github.com/landsman/betterstack-symfony/blob/33719a84cba1f82ac78c0b626be3a566dd7ebc09/src/Modules/Infrastructure/BetterStackHandler.php#L21
```php
$this->logtailHandler->pushProcessor(new PsrLogMessageProcessor());
```

I would also recommend to read https://peakd.com/hive-168588/@crell/using-psr-3-placeholders-properly

**Reply by landsman** [reply]
It works, thanks!

I understand that this behaviour can be useful for some cases, but for me it will be mostly about reading it as stupid log file than anything else.
